moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -43,34 +43,49 @@
 #include <geometry_msgs/msg/pose_stamped.h>
 #include <moveit/robot_state/conversions.h>
 #include <moveit/utils/moveit_error_code.h>
+#include <moveit/planning_interface/planning_response.h>
 
 namespace moveit_cpp
 {
 MOVEIT_CLASS_FORWARD(PlanningComponent);  // Defines PlanningComponentPtr, ConstPtr, WeakPtr... etc
 
+/// \brief A function to choose the solution with the shortest path from a vector of solutions
+planning_interface::MotionPlanResponse
+getShortestSolution(std::vector<planning_interface::MotionPlanResponse> const& solutions);
```

Comment:
This looks like a good opportunity to start a new header which defines PlanSolutions and free functions related to it.